### ISSUE TITLE
lab20

### ISSUE THREAD

**Issue by elixir123102030** [elixir123102030/week4#5]
import pandas as pd
import numpy as np
from sklearn.preprocessing import OneHotEncoder, StandardScaler

data = pd.read_csv('/mnt/data/day20_integration.csv')

encoder = OneHotEncoder(sparse=False)
cat_columns = ['city', 'device_type']
encoded_data = encoder.fit_transform(data[cat_columns])

scaler = StandardScaler()
num_columns = ['pages_viewed', 'session_minutes', 'basket_value']
scaled_data = scaler.fit_transform(data[num_columns])

encoded_df = pd.DataFrame(encoded_data, columns=encoder.get_feature_names_out(cat_columns))
scaled_df = pd.DataFrame(scaled_data, columns=num_columns)
processed_data = pd.concat([encoded_df, scaled_df], axis=1)

processed_data['interaction_feature'] = processed_data['pages_viewed'] * processed_data['session_minutes']

original_stats = data.describe()
engineered_stats = processed_data.describe()

print("Original data statistics:")
print(original_stats)
print("\nEngineered data statistics:")
print(engineered_stats)
